sky/task.py

```diff
@@ -521,7 +522,43 @@ def from_yaml_config(
                 # If the src is not a str path, it is likely a dict. Try to
                 # parse storage object.
                 elif isinstance(src, dict):
-                    fm_storages.append((dst_path, src))
+                    if (src.get('store') ==
+                            storage_lib.StoreType.VOLUME.value.lower()):
+                        # Build the volumes config for resources.
+                        volume_config = {
+                            'path': dst_path,
+                        }
+                        if src.get('name'):
+                            volume_config['name'] = src.get('name')
+                        persistent = src.get('persistent', False)
+                        volume_config['auto_delete'] = not persistent
+                        volume_config_detail = src.get('config', {})
+                        if volume_config_detail:
+                            if volume_config_detail.get('size'):
+                                volume_config[
+                                    'disk_size'] = volume_config_detail.get(
+                                        'size')
+                            if volume_config_detail.get('storage_type'):
+                                volume_config[
+                                    'storage_type'] = volume_config_detail.get(
+                                        'storage_type')
+                            if volume_config_detail.get('disk_tier'):
+                                volume_config[
+                                    'disk_tier'] = volume_config_detail.get(
+                                        'disk_tier')
+                            if volume_config_detail.get('attach_mode'):
+                                volume_config[
+                                    'attach_mode'] = volume_config_detail.get(
+                                        'attach_mode')
```

Comment:
can we just apply the entire `volume_config_detail` to `volume_config`?

Reply:
Updated